CODE/Sources/PennyBOT/Services/CoinService.swift

```diff
@@ -2,32 +2,35 @@ import Foundation
 import AsyncHTTPClient
 import PennyModels
 import Logging
-//import PennyShared
 
-public struct CoinService {
+struct CoinService {
+    
+    enum ServiceError: Error {
+        case badStatus
+    }
     
     let logger: Logger
     let httpClient: HTTPClient
     
-    func postCoin(with coinRequest: CoinRequest) async throws -> String {
-        do {
-            let BASE_URL = ProcessInfo.processInfo.environment["API_BASE_URL"]
-            logger.trace("\(String(describing: BASE_URL))")
-            var request = HTTPClientRequest(url: "\(BASE_URL ?? "")/coin")
-            request.method = .POST
-            request.headers.add(name: "Content-Type", value: "application/json")
-            request.body = .bytes(Data(try! JSONEncoder().encode(coinRequest)))
-            
-            let response = try await httpClient.execute(request, timeout: .seconds(30), logger: self.logger)
-            logger.trace("HTTP head \(response)")
-            let body = try await response.body.collect(upTo: 1024 * 1024)
-            
-            let reply = String(buffer: body)
-            return reply
-            
-        } catch {
-            return "request failed: \(error)"
+    func postCoin(with coinRequest: CoinRequest) async throws -> CoinResponse {
+        let BASE_URL = ProcessInfo.processInfo.environment["API_BASE_URL"]
+        logger.trace("\(String(describing: BASE_URL))")
+        var request = HTTPClientRequest(url: "\(BASE_URL ?? "")/coin")
+        request.method = .POST
+        request.headers.add(name: "Content-Type", value: "application/json")
+        request.body = .bytes(Data(try! JSONEncoder().encode(coinRequest)))
```

Comment:
Let's remove the force unwrap here, _just in case_

Reply:
yeah I saw that afterwards too. 
I don't know how that could fail, but at the same time the function is throwing already, so no need to risk anything.